lib/pbench/test/unit/server/test_cache_manager.py

```diff
@@ -485,260 +499,127 @@ def mock_run(args, **kwargs):
-                }
-            }
-            """
-
-            m.setattr(Tarball, "__init__", TestCacheManager.MockTarball.__init__)
-            m.setattr(Controller, "__init__", TestCacheManager.MockController.__init__)
-            tb = Tarball(tar, Controller(Path("/mock/archive"), cache, None))
-            tar_dir = TestCacheManager.MockTarball.create_tarball("tar_name")
-            tb.cache_map(tar_dir)
-            assert tb.cachemap["tar_name"]["details"].type == "directory"
+            assert c_map["details"].type == "symlink"
+            assert c_map["details"].name == "f1112_sym"
 
             # test get_info with a file path
-            file_info = tb.get_info(Path("tar_name/subdir1/f1.txt"))
+            file_info = tb.get_info(Path("dir_name/subdir1/f11.txt"))
             expected_info = {
-                "location": Path("/tmp/tar_name/subdir1/f1.txt"),
-                "name": "f1.txt",
-                "size": "1024",
+                "location": Path(f"{tar_dir}/subdir1/f11.txt"),
+                "name": "f11.txt",
+                "resolve_path": Path(f"{tar_dir}/subdir1/f11.txt"),
+                "resolve_type": None,
+                "size": 0,
                 "type": "file",
             }
             assert file_info == expected_info
 
             # test get_info with a directory path
-            dir_info = tb.get_info(Path("tar_name/subdir1"))
+            dir_info = tb.get_info(Path("dir_name/subdir1"))
             expected_info = {
                 "directory": ["subdir13", "subdir12", "subdir11"],
-                "file": ["f1.txt"],
+                "file": ["f11.txt"],
+                "location": Path(f"{tar_dir}/subdir1"),
+                "name": "subdir1",
+                "resolve_path": Path(f"{tar_dir}/subdir1"),
+                "resolve_type": None,
+                "size": None,
+                "type": "directory",
             }
             assert dir_info == expected_info
 
             # test get_info with an empty directory path
-            dir_info = tb.get_info(Path("tar_name/subdir1/subdir11"))
+            dir_info = tb.get_info(Path("dir_name/subdir1/subdir11"))
             expected_info = {
-                "location": Path("/tmp/tar_name/subdir1/subdir11"),
+                "directory": [],
+                "file": [],
+                "location": Path(f"{tar_dir}/subdir1/subdir11"),
                 "name": "subdir11",
+                "resolve_path": Path(f"{tar_dir}/subdir1/subdir11"),
+                "resolve_type": None,
+                "size": None,
                 "type": "directory",
             }
             assert dir_info == expected_info
 
             # test get_info with a symlink path
-
-            # Note: Need to figure out symlink issue
-            link_info = tb.get_info(Path("tar_name/subdir1/subdir13/subdir111/f1_sym"))
+            link_info = tb.get_info(
+                Path("/dir_name/subdir1/subdir13/subdir111/f1112_sym")
+            )
```

Comment:
The issue is that the argument to `get_info()` is supposed to be a path _within the unpacked tarball_, right?  Such a path should be relative, not absolute.  (And, it should have nothing to do with `tmp_path`.)

_**IF**_ the tarball were to contain absolute paths, they would be references to the the file system on the host where the result was collected, so they would be largely meaningless in the cache map or the file system where the cache resides.  So, if someone requests an item from the cache with an absolute path, we cannot simply dive into the cache map -- we first have to remove the leading components from the requested path until we reach the start of the unpacked contents.  (I was naively thinking that, because the tarball was created at `/mock/dir_name.tar.xz`, then the results that it contains must be in `/mock/dir_name/...`, and so the path in the cachemap would start with `dir_name` (_with no leading slash_).)

So, please remove the leading slash from this test.  If we want to support paths with leading slashes, those should be tested separately (and, I'm becoming increasingly convinced that we should _not_ support absolute references).